File: 투 포인터/2018.cpp
#include<iostream>

using namespace std;

int main(){
    int n, ans=1; // 자기 자신 포함
    cin>>n;
    int start=1, end=1; // 포인터
    int sum =1; // start부터 end까지의 합
    while(end<n){
        if(sum<=n){
            if(sum==n) ans++;
            end++;
            sum+=end;
        }
        else{
            sum-=start;
            start++;
        }
    }
    cout<<ans;
}

File: 투 포인터/2018.java
import java.util.*;

public class N006 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int p1 = 1;
        int p2 = 1;
        int sum = 1;
        int ans = 0;
        while(p2 < n){
            if(sum==n) {
                ans++;
                p2++;
                sum += p2;
            }
            else if(sum<n){
                p2++;
                sum += p2;
            }
            else {
                sum -= p1;
                p1++;
            }
        }
        System.out.println(ans+1);
    }
}
